pre-sync/oci-image-verification/README.md

```diff
@@ -0,0 +1,156 @@
+# Sample OCI Image Signature Verification with Config Sync
+
+This sample demonstrates how to verify the signature of an OCI image that Config
+Sync is managing. The webhook monitors the `configsync.gke.io/image-to-sync`
+annotation on specified RootSync or RepoSync resources, which contains the full
+URL of the image to sync. By comparing the previous and updated values in the
+admission review request, the webhook can detect if a new image has been
+introduced, triggering an image verification process.
+
+## Prerequisites
+
+CLI: [OpenSSL], [Cosign], [Docker], [Kubectl], [Crane]
+
+Environment: GKE cluster, Google Cloud Artifact Registry.
+
+## Build the signature verification server
+
+```shell
+docker build -t <IMAGE_REGISTRY_URL>:latest . && docker push <IMAGE_REGISTRY_URL>:latest
+```
+
+## Create a Namespace and Kubernetes Service Account
+
+This is intended to group everything under the same namespace. In this example, `signature-verification` is used as a reference:
+
+```shell
+kubectl create ns signature-verification
+```
+
+It is recommended to create the Kubernetes Service Account in advance to facilitate authentication in subsequent steps:
+
+```shell
+kubectl create sa signature-verification-sa -n signature-verification
+```
+
+## Authentications
+
+### Image registry authentication
+
+This example demonstrates how to authenticate the Cosign client within the
+admission webhook using built-in Google authentication to access the source
+image repository. You can adapt this authentication method to suit your specific
+verification client and registry.
+
+### IAM setup for the signature verification server
+
+- Give the KSA permission to read images from source Artifact Registry
+```shell
+gcloud artifacts repositories add-iam-policy-binding <SOURCE_IMAGE_AR_REPO> \
+  --location=<LOCATION> \
+  --member="serviceAccount:<PROJECT_ID>.svc.id.goog[signature-verification/signature-verification-sa]" \
+  --role=roles/artifactregistry.reader \
+  --project=peip-monitor
```

Comment:
Suggest not using a personal project, but using `YOUR_PROJECT_ID`.

Reply:
Done